controller/control.go

```diff
@@ -134,9 +135,33 @@ func (c *Controller) UpdateVolStatus() {
 	} else {
 		c.ReadOnly = true
 	}
+	c.RWReplicaCount = rwReplicaCount
 
-	logrus.Infof("Previously Volume RO: %v, Currently: %v,  Total Replicas: %v,  RW replicas: %v, Total backends: %v",
-		prev, c.ReadOnly, len(c.replicas), rwReplicaCount, len(c.backend.backends))
+	logrus.Infof("Previously Volume RO: %v, Currently: %v, Total Replicas: %v, RW replicas: %v, Total backends: %v",
+		prevState, c.ReadOnly, len(c.replicas), rwReplicaCount, len(c.backend.backends))
+}
+
+func (c *Controller) UpdateCheckpoint() {
+	var checkpoint string
+	var rwReplicaCount int
+	prevCheckpoint := c.Checkpoint
+	for _, replica := range c.replicas {
+		if replica.Mode == "RW" {
+			rwReplicaCount++
+		}
+	}
+
+	if rwReplicaCount == c.ReplicationFactor {
+		latestSnapshot, err := c.backend.GetLatestSnapshot()
+		if err == nil {
+			if err := c.backend.SetCheckpoint(latestSnapshot); err == nil {
+				checkpoint = latestSnapshot
+			}
+		}
+	}
+	c.Checkpoint = checkpoint
```

Comment:
if 'err' happens in any of the above statement, c.Checkpoint gets set to "".. is that the expected?
can you put the comments for the function with i/p, o/p and the functionality?

Reply:
Yes that is the expectation. Adding comment for the same.